framework/src/utils/SlepcSupport.C

```diff
@@ -526,6 +526,32 @@ mooseSlepcEigenFormFunctionB(SNES snes, Vec x, Vec r, void * ctx)
   moosePetscSNESFormFunction(snes, x, r, ctx, Moose::KT_EIGEN);
   PetscFunctionReturn(0);
 }
+
+PetscErrorCode
+mooseSlepcEigenFormFunctionAB(SNES /*snes*/, Vec x, Vec Ax, Vec Bx, void * ctx)
+{
+  PetscFunctionBegin;
+
+  EigenProblem * eigen_problem = static_cast<EigenProblem *>(ctx);
+  NonlinearSystemBase & nl = eigen_problem->getNonlinearSystemBase();
+  System & sys = nl.system();
+
+  PetscVector<Number> X_global(x, sys.comm()), AX(Ax, sys.comm()), BX(Bx, sys.comm());
+
+  // update local solution
+  X_global.localize(*sys.current_local_solution.get());
+
+  AX.zero();
```

Comment:
These `zero` and `close` operations can be moved into `computeResidualABType`.

Reply:
I like to keep this way because it consistent with elsewhere.